src/scripts/Providers/OSUI/Utils/FloatingUI/FloatingUI.ts

```diff
@@ -0,0 +1,170 @@
+				}
+
+				// Check if allowedPlacements is empty. If it is, then add a default placement, as it is mandatory
+				if (this._floatingUIOptions.autoPlacementOptions.allowedPlacements.length <= 0) {
+					this._floatingUIOptions.autoPlacementOptions.allowedPlacements.push(
+						OSFramework.OSUI.GlobalEnum.FloatingPosition.BottomStart
+					);
+				}
+
+				_middlewareArray.push(window.FloatingUIDOM.autoPlacement(this._floatingUIOptions.autoPlacementOptions));
+				// Make sure to also use shift for better auto behaviour
+				_middlewareArray.push(window.FloatingUIDOM.shift());
+			}
+
+			// Add offset value to middleware
+			if (this._floatingUIOptions.position !== OSFramework.OSUI.GlobalEnum.FloatingPosition.Center) {
+				_middlewareArray.push(window.FloatingUIDOM.offset(this._getOffsetValue()));
+			}
+
+			// Set the computePosition method. This is the main provider method to set the balloon position
+			const _eventOnUpdatePosition = async () => {
+				try {
+					const { x, y } = await window.FloatingUIDOM.computePosition(
+						this._floatingUIOptions.anchorElem,
+						this._floatingUIOptions.floatingElem,
+						{
+							placement: this._floatingUIOptions.position,
+							middleware: _middlewareArray,
+						}
+					);
+
+					// Update the Balloon CSS Variables with the returned optimal x & y for position
+					OSFramework.OSUI.Helper.Dom.Styles.SetStyleAttribute(
+						this._floatingUIOptions.floatingElem,
+						Enum.CssCustomProperties.YPosition,
+						y + OSFramework.OSUI.GlobalEnum.Units.Pixel
+					);
+					OSFramework.OSUI.Helper.Dom.Styles.SetStyleAttribute(
+						this._floatingUIOptions.floatingElem,
+						Enum.CssCustomProperties.XPosition,
+						x + OSFramework.OSUI.GlobalEnum.Units.Pixel
+					);
+				} catch (error) {
+					throw new Error(Providers.OSUI.ErrorCodes.FloatingUI.FailCallProvider);
+				}
+			};
+
+			// Set the position
+			_eventOnUpdatePosition()
+				.then(() => {
+					// If updatePosition is used, set a callback to run autoUpdate method. This is also used to then clean-up on destroy
+					if (this._floatingUIOptions.updatePosition) {
+						this._eventOnUpdateCallback = window.FloatingUIDOM.autoUpdate(
+							this._floatingUIOptions.anchorElem,
+							this._floatingUIOptions.floatingElem,
+							_eventOnUpdatePosition.bind(this)
+						);
+					}
+				})
+				.catch((error) => {
+					console.error(
+						`${Providers.OSUI.ErrorCodes.FloatingUI.FailCallProvider}: FloatingUI failed to update the position: ${error}`
+					);
+				});
+		}
+
+		/**
+		 * Method to build the FloatingUI
+		 *
+		 * @memberof FloatingUI
+		 */
+		public build(): void {
+			this._setFloatingPosition();
+		}
+
+		/**
+		 * Method to run when the Ballon closes. This will clean listeners on the provider side and stop observing the target
+		 *
+		 * @memberof FloatingUI
+		 */
+		public close(): void {
+			this._eventOnUpdateCallback();
+			setTimeout(() => {
```

Comment:
For this, we can use the **OSFramework.OSUI.Helper.ApplySetTimeOut** ;)

Reply:
done